src/RWA007-A/scripts/RWA007Deployment.s.sol

```diff
@@ -0,0 +1,174 @@
+        }
+
+        // join it
+        address RWA_JOIN = getEnvAddress("RWA_JOIN");
+        if (RWA_JOIN == address(0)) {
+            RWA_JOIN =  JOIN_FAB.newAuthGemJoin(MCD_PAUSE_PROXY, ILK, RWA_TOKEN);
+        }
+
+        // route it
+        address RWA_OUTPUT_CONDUIT = getEnvAddress("RWA_OUTPUT_CONDUIT");
+        if (RWA_OUTPUT_CONDUIT == address(0)) {
+            RwaOutputConduit3 outputC =  new RwaOutputConduit3(MCD_PSM_USDC_A);
+            outputC.rely(MCD_PAUSE_PROXY);
+
+            RWA_OUTPUT_CONDUIT = address(outputC);
+        }
+        
+        // urn it
+        address RWA_URN = getEnvAddress("RWA_URN");
+        if (RWA_URN == address(0)) {
+            RwaUrn2 urn = new RwaUrn2(MCD_VAT, MCD_JUG, RWA_JOIN, MCD_JOIN_DAI, RWA_OUTPUT_CONDUIT);
+            urn.rely(MCD_PAUSE_PROXY);
+            urn.deny(msg.sender);
+
+            RWA_URN = address(urn);
+
+            // Set _quitTo address to the URN and deny deplyer
+            RwaOutputConduit3(RWA_OUTPUT_CONDUIT).file("quitTo", RWA_URN);
+            RwaOutputConduit3(RWA_OUTPUT_CONDUIT).deny(msg.sender);
+        }
+
+        // jar it
+        address RWA_JAR = getEnvAddress("RWA_JAR");
+        if (RWA_JAR == address(0)) {
+            RWA_JAR = address(new RwaJar(address(CHANGELOG)));
+        }
+
+        // route it JAR
+        address RWA_INPUT_CONDUIT_JAR = getEnvAddress("RWA_INPUT_CONDUIT_JAR");
+        if (RWA_INPUT_CONDUIT_JAR == address(0)) {
+            RwaInputConduit3 inputCJar = new RwaInputConduit3(MCD_PSM_USDC_A, RWA_JAR);
+            inputCJar.rely(MCD_PAUSE_PROXY);
+            inputCJar.deny(msg.sender);
+
+            RWA_INPUT_CONDUIT_JAR = address(inputCJar);
+        }
+
+        // route it URN
+        address RWA_INPUT_CONDUIT_URN = getEnvAddress("RWA_INPUT_CONDUIT_URN");
+        if (RWA_INPUT_CONDUIT_URN == address(0)) {
+            RwaInputConduit3 inputCUrn = new RwaInputConduit3(MCD_PSM_USDC_A, RWA_URN);
+            inputCUrn.rely(MCD_PAUSE_PROXY);
+            inputCUrn.deny(msg.sender);
+
+            RWA_INPUT_CONDUIT_URN = address(inputCUrn);
+        }
+        
+        vm.stopBroadcast();
+
+        logJSONTuple("SYMBOL", SYMBOL);
+        logJSONTuple("NAME", NAME);
+        logJSONTuple("ILK", string(abi.encodePacked(SYMBOL, string("-"), LETTER)));
+        logJSONTuple(SYMBOL, vm.toString(RWA_TOKEN));
+        logJSONTuple(concatString("MCD_JOIN_", SYMBOL_LETTER), vm.toString(RWA_JOIN));
+        logJSONTuple(concatString(SYMBOL_LETTER, "_URN"), vm.toString(RWA_URN));
+        logJSONTuple(concatString(SYMBOL_LETTER, "_JAR"), vm.toString(RWA_JAR));
+        logJSONTuple(concatString(SYMBOL_LETTER, "_OUTPUT_CONDUIT"), vm.toString(RWA_OUTPUT_CONDUIT));
+        logJSONTuple(concatString(SYMBOL_LETTER, "_INPUT_CONDUIT_URN"), vm.toString(RWA_INPUT_CONDUIT_URN));
+        logJSONTuple(concatString(SYMBOL_LETTER, "_INPUT_CONDUIT_JAR"), vm.toString(RWA_INPUT_CONDUIT_JAR));
```

Comment:
If you add a logJSONTuple(string memory, address) overload, you could remove all these vm.toString() calls here and abstract that away.

Reply:
done